docs/llvm/pass/hello-world.md

```diff
@@ -0,0 +1,130 @@
+# Write a Hello World Pass
+
+In this section, we guide you through writing and running a simple pass that prints the names of all functions contained in the input IR module.
+
+## Setup the Build
+
+Let's assume that you have cloned the LLVM monorepo and setup the build directory at `build/`, and the current working directory is the root of the monorepo. In this section, we will build the new pass in-tree, which means that the source tree of the new pass is directly embedded in the source tree of the LLVM monorepo.
```

Comment:
> you have cloned the LLVM monorepo and setup the build directory

Links to github llvm monorepo

`[LLVM monorepo](https://github.com/llvm/llvm-project)`

> setup the build directory at `build/`

how to build? (footnote)?

```diff
- Let's assume that you have cloned the LLVM monorepo and setup the build directory at `build/`
+ Let's assume that you have cloned the LLVM monorepo and setup the build directory at `build/` [^howtobuild]

[^howtobuild]: \ links to official guidance
```

> will build the new pass in-tree

How about out-of-tree passes? (footnote / links to related tutorial?)

Reply:
Seems that out-of-tree passes with the new PM do not have official documentation. Maybe it should included in the wiki (as a separated section).